tests/test_repository.py

```diff
@@ -3,315 +3,353 @@
+        return _build_script.relative_to(internal.SRC_DIR / repo.name)
 
-def test_remove_module_lines():
+    @pytest.fixture()
+    def modules(self):
+        """Return a list of modules for testing."""
+        return ["foo", "bar"]
+
+    def test_build_command_execution(
+        self, repo, mock_subprocess_handler, build_script, modules
+    ):
+        """Success case: execute the build command for a custom build script."""
+        repo.build_script = str(build_script)
+        repo.custom_build(modules)
+        assert "./tmp-build.sh" in mock_subprocess_handler.commands
+
+    def test_modules_loaded_at_runtime(
+        self, repo, mock_environment_modules_handler, build_script, modules
+    ):
+        """Success case: test modules are loaded at runtime."""
+        repo.build_script = str(build_script)
+        repo.custom_build(modules)
+        assert (
+            "module load " + " ".join(modules)
+        ) in mock_environment_modules_handler.commands
+        assert (
+            "module unload " + " ".join(modules)
+        ) in mock_environment_modules_handler.commands
+
+    # TODO(Sean) fix for issue https://github.com/CABLE-LSM/benchcab/issues/162
+    @pytest.mark.skip(
+        reason="""This will always fail since `parametrize()` parameters are
+        dependent on the `mock_cwd` fixture."""
+    )
+    @pytest.mark.parametrize(
+        ("verbosity", "expected"),
+        [
+            (
+                False,
+                "",
+            ),
+            (
+                True,
+                "Copying src/trunk/my-custom-build.sh to src/trunk/tmp-build.sh\n"
+                "chmod +x src/trunk/tmp-build.sh\n"
+                "Modifying tmp-build.sh: remove lines that call environment "
+                "modules\n"
+                "Loading modules: foo bar\n"
+                "Unloading modules: foo bar\n",
+            ),
+        ],
+    )
+    def test_standard_output(self, repo, build_script, modules, verbosity, expected):
+        """Success case: test non-verbose standard output for a custom build script."""
+        repo.build_script = str(build_script)
+        with contextlib.redirect_stdout(io.StringIO()) as buf:
+            repo.custom_build(modules, verbose=verbosity)
+        assert buf.getvalue() == expected
+
+    def test_file_not_found_exception(self, repo, build_script, modules, mock_cwd):
+        """Failure case: cannot find custom build script."""
+        build_script_path = mock_cwd / internal.SRC_DIR / repo.name / build_script
+        build_script_path.unlink()
+        repo.build_script = str(build_script)
+        with pytest.raises(
+            FileNotFoundError,
+            match=f"The build script, {build_script_path}, could not be "
+            "found. Do you need to specify a different build script with the 'build_script' "
+            "option in config.yaml?",
+        ):
+            repo.custom_build(modules)
+
+
+class TestRemoveModuleLines:
```

Comment:
Too bad this is preventing the `repo` fixture from being on auto-use. It's probably a sign this function should be moved to `utils`. But that's another issue.

Reply:
I'm not sure if `repo` should be an `autouse` fixture since the tests need the `CableRepository` instance that it returns for testing, i.e. we would have to pass it in as an argument to each test regardless.

But yep `remove_module_lines` is probably in the wrong module but given it is only used in this file I haven't bothered to move it yet